lib/kubernetes-deploy/render_task.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+require 'tempfile'
+
+require 'kubernetes-deploy/renderer'
+require 'kubernetes-deploy/template_discovery'
+
+module KubernetesDeploy
+  class RenderTask
+    def initialize(logger:, current_sha:, template_dir:, bindings:)
+      @logger = logger
+      @template_dir = template_dir
+      @renderer = KubernetesDeploy::Renderer.new(
+        current_sha: current_sha,
+        bindings: bindings,
+        template_dir: @template_dir,
+        logger: @logger,
+      )
+    end
+
+    def run(*args)
+      run!(*args)
+      true
+    rescue KubernetesDeploy::InvalidTemplateError, KubernetesDeploy::TaskConfigurationError
+      @logger.print_summary(:failure)
+      false
+    end
+
+    def run!(filenames, stream)
+      if filenames.empty?
+        filenames = TemplateDiscovery.new(@template_dir).templates
+        if filenames.empty?
+          message = "no templates found in template dir #{@template_dir}"
+          @logger.summary.add_action("Configuration invalid: #{message}")
+          raise KubernetesDeploy::TaskConfigurationError, message
+        end
+      end
+
+      validate_configuration(filenames)
+
+      filenames.each { |filename| render_filename(filename, stream) }
+
+      @logger.print_summary(:success)
+    end
+
+    private
+
+    def render_filename(filename, stream)
+      file_content = File.read(File.join(@template_dir, filename))
+      rendered_content = @renderer.render_template(filename, file_content)
+      YAML.load_stream(rendered_content) do |doc|
+        stream.puts YAML.dump(doc)
```

Comment:
Currently, if you are rendering multiple templates and one is invalid, we're going to output results until we hit the bad one, and if you're running this manually, those templates will be visually interspersed with our stderr logging. Is that what we want, or should we save up the results and dump them all to stdout on successful completion? I'm thinking the latter would be better UX.

Reply:
I agree w/ @KnVerey here. Its really frustrating to have to fix items 1 at a time.